core/unit_test/CMakeLists.txt

```diff
@@ -338,6 +339,35 @@ KOKKOS_ADD_ADVANCED_TEST( UnitTest_PushFinalizeHook_terminate
     ALWAYS_FAIL_ON_ZERO_RETURN
 )
 
+if(Kokkos_ENABLE_PROFILING)
+  add_library(
+     printer-tool SHARED
```

Comment:
If this is intended to be a loadable object (rather than linked against), I believe you want to use `MODULE` instead of `SHARED`.

Reply:
Done, thanks